src/DynamoPublish/ShareWorkspaceClient.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Greg;
+using Greg.Requests;
+using Greg.Responses;
+using RestSharp;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Dynamo.Publish
+{
+    public class ShareWorkspaceClient
+    {
+        private IAuthProvider _authenticationProvider;
+        private string _baseUrl;
+        private IGregClient _client;
+
+        public IAuthProvider AuthenticationProvider
+        {
+            get { return _authenticationProvider; }
+        }
+
+        public string BaseUrl
+        {
+            get { return _baseUrl; }
+        }
+
+        public ShareWorkspaceClient(string baseUrl, IAuthProvider authProvider)
+        {
+            _baseUrl = baseUrl;
+            _authenticationProvider = authProvider;
+            _client = new GregClient(_authenticationProvider, _baseUrl);
```

Comment:
Hi @pbidenko, we need to add a similar endpoint to flood server as we have in the package manager. Consider the possibility that their respect TOU's change over time.

So, we'll need to a flood PR here that stores the TOU acceptance in MongoDB. Please use the same pattern as the one established there...